python/pyarrow/_parquet.pyx

```diff
@@ -493,6 +493,16 @@ cdef class ColumnChunkMetaData(_Weakrefable):
         """Uncompressed size in bytes (int)."""
         return self.metadata.total_uncompressed_size()
 
+    @property
+    def has_offset_index(self):
+        """Has offset index"""
```

Comment:
A actual question for this PR: we already do have a `has_index_page` attribute. How does that exactly differ?

Reply:
Also it would be nice to implement `has_index_page` and `index_page_offset`...

Reply:
To be short, `IndexPage` is not `PageIndex`.

1. https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L536 Currently, index page turns to be something that user defined.
2. PageIndex is a zonemap for pages

So they are different things

Reply:
Ahah, I forgot about that.